packages/alfa-rules/src/sia-er65/diagnostics.ts

```diff
@@ -0,0 +1,60 @@
+import { Diagnostic } from "@siteimprove/alfa-act";
+
+export type Reason = "auto-detected" | "user-answered" | "good-class";
+
+/**
+ * @internal
+ */
+export class ExtendedDiagnostic extends Diagnostic {
+  public static of(
```

Comment:
To make the reason not (really) an optional argument, we can use an overload and default to `Diagnostic` when none is provided. See #1036.